config/routes.rb

```diff
@@ -0,0 +1,20 @@
+# For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+Rails.application.routes.draw do
+
+  root 'works#main'
+
+  resources :works do
+    resources :votes, only: [:create]
+  end
+
+  # TODO Write this nested route - if you are not logged in you cannot Upvote
+  get '/works/:work_id/upvote', to: 'votes#create', as: 'upvote'
```

Comment:
You have two routes here that go to `votes#create`, but your application only uses one of them.